fabtests/pytest/efa/test_rdm.py

```diff
@@ -4,11 +4,18 @@
 
 def run_rdm_test(cmdline_args, executable, iteration_type, completion_type, memory_type, message_size):
     from common import ClientServerTest
+    # It is observed that cuda_to_cuda tests requires larger time-out limit (~240 secs) to test all
+    # message sizes, on p4d instances.
```

Comment:
on all build types or just certain build type?

Reply:
because for release build 240 is a long time.

Reply:
It is for certain builds: debug and mem-poisoning.

Reply:
I updated the comment. I also found `host_to_cuda` and `cuda_to_host` test also requires such larger timeout for debug and mem-poisoning builds. So I updated this PR to increase timeout for all cuda tests when testing all msg sizes.

Reply:
Can you check the instance type from pytest?

Reply:
You can check the instance type by pulling the ec2 metadata. But I don't think that check is desirable as this behavior may apply to other GDRDMA enabled instances as well in the future. Also this larger time cost is only observed on debug and mem-poisoning builds of libfabric, which is not straightforward to check inside fabtests. Increasing the timeout limit for gpu tests in general makes the test more flexible.